cryptocurrency-subdomain/src/main/kotlin/com/ajax/cryptocurrency/application/ports/kafka/KafkaProducer.kt

```diff
@@ -0,0 +1,9 @@
+package com.ajax.cryptocurrency.application.ports.kafka
+
+import com.ajax.cryptocurrency.domain.CryptocurrencyDomain
+
+interface KafkaProducer<T>{
```

Comment:
NotificationPublisher

fun publishNotification

these names are general and describe what's happening. and implementation can publish to kafka or any other broker

Reply:
renamed

Reply:
This interface you can leave since it is used by another infrastructure component. Consumer was not used anywhere